api/metrics-api-util/src/main/java/org/hawkular/metrics/api/jaxrs/config/ConfigurationKey.java

```diff
@@ -88,6 +88,9 @@
     INGEST_MAX_RETRIES("hawkular.metrics.ingestion.retry.max-retries", null, "INGEST_MAX_RETRIES", false),
     INGEST_MAX_RETRY_DELAY("hawkular.metrics.ingestion.retry.max-delay", null, "INGEST_MAX_RETRY_DELAY", false),
 
+    VERSION_CHECK_DELAY("hawkular.metrics.version.check.delay", "10", "VERSION_CHECK_DELAY", false),
```

Comment:
..unit is..? seconds?

Reply:
Thread.sleep indicates milliseconds though.. I would increase this value and decrease the max retries.

Reply:
The value is in seconds. I could rename the vars to something like hawkular.metrics.version.check.delay-sec and VERSION_CHECK_DELAY_SEC. I thought a bit about what to use for defaults. The key for me is that I settled on a overall time of 3 minutes to way for schema updates to finish. If after 3 minutes schema updates are not finishes, there will be a container restart for hawkular-metrics. 

I don't want to go to high for the delay because I think upgrades are going to be quite common. In most upgrade scenarios we only apply a relatively small amount of schema changes that should complete quickly.